packages/flutter/test/animation/animation_controller_test.dart

```diff
@@ -396,6 +396,45 @@ void main() {
     expect(controller.value, 1.0);
   });
 
+  test('resetting animation works at all phases', (){
+    final List<AnimationStatus> statusLog = <AnimationStatus>[];
+    final AnimationController controller = new AnimationController(
+      duration: const Duration(milliseconds: 100),
+      value: 0.0,
+      lowerBound: 0.0,
+      upperBound: 1.0,
+      vsync: const TestVSync(),
+    )..addStatusListener(statusLog.add);
+
+    expect(controller.value, 0.0);
+    expect(controller.status, AnimationStatus.dismissed);
+
+    controller.reset();
+
+    expect(controller.value, 0.0);
+    expect(controller.status, AnimationStatus.dismissed);
+
+    statusLog.clear();
+    controller.forward();
+    tick(const Duration(milliseconds: 0));
+    tick(const Duration(milliseconds: 50));
+    controller.reset();
```

Comment:
maybe add an expectation after each tick to verify that the controller is in the state we expect?